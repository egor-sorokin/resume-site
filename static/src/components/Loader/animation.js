import { Power2 } from 'gsap';


export const attachHideAnimation = (targetTween) => {
  targetTween
    .to('body', 0, { overflowY: 'hidden' })
    .to(
      '.loader__inner', 0.1,
      { opacity: 0, ease: Power2.easeOut },
    )
    .to('body', 0, { overflowY: 'initial' });
};
