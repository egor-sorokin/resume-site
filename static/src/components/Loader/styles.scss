@import '../../styles/base/_vars.scss';

.loader {
  width: 100vw;
  height: 100vh;

  &__inner {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    height: 100%;
  }
}
