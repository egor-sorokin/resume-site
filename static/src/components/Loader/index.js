import React from 'react';
import './styles.scss';

const Loader = () => (
  <div className="loader">
    <h3 className="loader__inner font-s-20 text-c-l-dune">
      Loading...
    </h3>
  </div>
);

export default Loader;
