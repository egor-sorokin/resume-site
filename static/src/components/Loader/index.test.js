// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Loader from './index';

describe('<Loader />', () => {
  let component;
  let spy;
  const tweenline = {
    play: () => {
    }
  };

  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Loader />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 component', () => {
      expect(component.children()).toHaveLength(1);
    });
  });


  describe('lifecycle tests', () => {
    afterEach(() => {
      spy.mockClear();
    });


    it('calls componentDidMount', () => {
      spy = jest.spyOn(Loader.prototype, 'componentDidMount');
      // eslint-disable-next-line
      component = mount(<Loader />);
      expect(spy).toHaveBeenCalledTimes(1);
    });


    it('calls componentWillUnmount', () => {
      spy = jest.spyOn(Loader.prototype, 'componentWillUnmount');
      // eslint-disable-next-line
      component = mount(<Loader />);
      component.unmount();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });


  describe('inner functions tests', () => {
    afterEach(() => {
      spy.mockClear();
    });


    it('calls play an animation', () => {
      spy = jest.spyOn(tweenline, 'play');
      // eslint-disable-next-line
      component = mount(<Loader />);
      component.setState({loaderTween: tweenline});
      component.state().loaderTween.play();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
