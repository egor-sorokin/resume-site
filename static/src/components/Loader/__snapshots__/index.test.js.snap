// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> base tests matches snapshot 1`] = `
<div
  className="loader"
>
  <h3
    className="loader__inner font-s-20 text-c-l-dune"
  >
    Loading...
  </h3>
</div>
`;
